x-pack/plugins/infra/public/pages/metrics/hosts/hooks/use_data_view.ts

```diff
@@ -6,89 +6,75 @@
  */
 
 import { i18n } from '@kbn/i18n';
-import { useCallback, useState, useEffect, useMemo } from 'react';
+import { v5 as uuidv5 } from 'uuid';
+import { useEffect, useMemo, useState } from 'react';
 import { useKibana } from '@kbn/kibana-react-plugin/public';
 import createContainer from 'constate';
-import type { DataView } from '@kbn/data-views-plugin/public';
-import { InfraClientStartDeps } from '../../../../types';
+import type { DataView, DataViewSpec } from '@kbn/data-views-plugin/public';
 import { useTrackedPromise } from '../../../../utils/use_tracked_promise';
+import type { InfraClientStartDeps } from '../../../../types';
+import { DATA_VIEW_PREFIX, TIMESTAMP_FIELD } from '../constants';
+
+export const generateDataViewId = (indexPattern: string) => {
+  // generates a unique but the same uuid as long as the index pattern doesn't change
+  return `${DATA_VIEW_PREFIX}_${uuidv5(indexPattern, uuidv5.DNS)}`;
+};
 
 export const useDataView = ({ metricAlias }: { metricAlias: string }) => {
-  const [metricsDataView, setMetricsDataView] = useState<DataView>();
   const {
     services: { dataViews, notifications },
   } = useKibana<InfraClientStartDeps>();
 
-  const [createDataViewRequest, createDataView] = useTrackedPromise(
+  const [dataView, setDataView] = useState<DataView>();
+  const [hasError, setHasError] = useState<boolean>(false);
+
+  const [createAdhocDataViewRequest, createAdhocDataView] = useTrackedPromise(
```

Comment:
**question:** I see you initially tried switching to `useAsync` for handling this logic, but then you switched back to the current implementation, there is any reason why you couldn't keep that abstraction?

Reply:
I can't exactly remember. It worked mostly fine. I was using it with `useTrackedPromise`. It would only make sense if we stop using `useTrackedPromise`. With it, I didn't see much value in adding another layer of abstraction. 

Do you have a strong preference for using `useAsync` in this case? I'm open to changing it if it makes sense.

Reply:
Not particularly a preference, I guess it just makes it easier to read and configure the async operation, but it should be equivalent. Don't worry about this right now and go on, I believe would be nice one day to speak on how we can unify the server-state management, maybe integrating with existing solutions (react-query, swr, etc...)